### PR TITLE
runfix: Respect deviceId in calls, use selfuser from avs callbacks (WEBAPP-6880)

### PR DESCRIPTION
**Pull request by AndyLnd**

### CODE REVIEW

File: src/script/calling/Call.ts [hunk 1]
```diff
@@ -50,17 +49,23 @@ export class Call {
...
+  }
+
   addParticipant(participant: Participant): void {
-    this.participants.unshift(participant);
+    this.participants.push(participant);
   }
 
-  getParticipant(userId: string): Participant {
-    return this.participants().find(participant => participant.userId === userId);
+  getParticipant(userId: UserId, deviceId: DeviceId): Participant {
```
Comment (bennycode) [inline src/script/calling/Call.ts]: Let's call it `clientId`.

File: test/unit_tests/components/list/conversationListCallingCellSpec.js [hunk 2]
```diff
@@ -27,9 +27,9 @@ import 'src/script/components/list/conversationListCallingCell';
 import {User} from 'src/script/entity/User';
 
 function createCall(state) {
-  const call = new Call();
+  const selfParticipant = new Participant('selfId');
+  const call = new Call('', '', undefined, selfParticipant);
```
Comment (bennycode) [inline test/unit_tests/components/list/conversationListCallingCellSpec.js:32]: Let's use `createRandomUuid` to pass IDs to the call so it looks a little more like on production.

File: src/script/calling/Call.ts (hunk 1 above)
Comment (AndyLnd) [reply src/script/calling/Call.ts]: I named it `deviceId` here, as the property on the `Participant` is also called `deviceId`

Comment (bennycode) [reply src/script/calling/Call.ts]: So `DeviceId` should also be renamed.

Comment (AndyLnd) [reply src/script/calling/Call.ts]: renamed everywhere

**Pull request merged**